py/desispec/zcatalog.py

```diff
@@ -178,11 +180,24 @@ def create_summary_catalog(specgroup, indir=None, specprod=None):
     specgroup : str
         The option to run the code on ztile* files or zpix* files.
         It can either be 'zpix' or 'ztile'
-    indir : str
+    indir : str, optional
         Input directory to look for zpix/ztile files.
-    specprod : str
+    specprod : str, optional
         Internal Release Name for the DESI Spectral Release.
         Used to derive input directory if indir is not provided.
+    order_columns : bool, optional
+        If ``False``, do not try to reorder the columns in the final file. Default is ``True``.
+    all_columns : bool, optional
+        Whether or not to include all the columns into the final table. Default is ``True``.
+    columns_list : list, optional
+        If all_columns is ``False``, list of columns to include in the final table.
+        If None, a list of pre-decided summary columns will be used. Default is ``None``.
+        The 'SV/MAIN' primary flag columns as well as the primary flag columns for the entire
+        catalog witll be included.
```

Comment:
minor: typo "witll" -> "will"

Suggestion: let's add an explicit check + informative error message`if all_columns and columns_list is not None: ...`.  i.e. if the user supplies `columns_list` they might not notice that they also should set `all_columns=False`. Or otherwise let's merge this into a single option with something like
* `columns_list=None` (default) -> include all columns.
* `columns_list='minimal'` -> include a pre-defined set of minimal columns.
* `columns_list=list_of_strings` -> include only those columns.

Multi-type options are messy, but it avoids the alternate mess of setting one option requiring setting a different option to a specific value.